components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -101,14 +101,20 @@ public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx)
 
         String consumerKey = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getClientId();
         String authorizedUser = tokReqMsgCtx.getAuthorizedUser().toString();
-        int tenantId = OAuth2Util.getTenantId(tokReqMsgCtx.getOauth2AccessTokenReqDTO().getTenantDomain());
+
         boolean isUsernameCaseSensitive = IdentityUtil.isUserStoreInUsernameCaseSensitive(authorizedUser);
-        SpOAuth2ExpiryTimeConfiguration spTimeConfigObj = OAuth2Util.getSpTokenExpiryTimeConfig(consumerKey, tenantId);
+        // loading the stored application data
+        OAuthAppDO oAuthAppDO = null;
+        try {
+            oAuthAppDO = OAuth2Util.getAppInformationByClientId(consumerKey);
+        } catch (InvalidOAuthClientException e) {
+            throw new IdentityOAuth2Exception("Error while retrieving app information for clientId : " + consumerKey, e);
+        }
         if (log.isDebugEnabled()) {
             log.debug("Service Provider specific expiry time enabled for application : " + consumerKey +
-                    ". Application access token expiry time : " + spTimeConfigObj.getApplicationAccessTokenExpiryTime()
-                    + ", User access token expiry time : " + spTimeConfigObj.getUserAccessTokenExpiryTime() +
-                    ", Refresh token expiry time : " + spTimeConfigObj.getRefreshTokenExpiryTime());
+                    ". Application access token expiry time : " + oAuthAppDO.getApplicationAccessTokenExpiryTime()
```

Comment:
Worth mentioning the time unit in the debug logs.

Reply:
Fix with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/477